### PR TITLE
Reenable disabled tests by pr body

### LINKED ISSUE #103789: Please ignore, issue opened for testing
issue opened for testing purposes

### LINKED ISSUE #103798: Small type misalignments in `nn.functional`
### 🐛 Describe the bug

This is a sub-task of #103761.

This is not user-facing as this only involves functions prefixed with an underscore.

---

The `_list_with_default` is used as this, where `input.size()` is `torch.Size`:
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/functional.py#L1135
However, it's annotated to accept `List` only. It's even possible to work on `int`:
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/modules/utils.py#L33-L35

---

Similar problem with `_verify_batch_size`
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/functional.py#L2481
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/functional.py#L2435

---

This `_in_projection_packed` is annotated to return a `List`:
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/functional.py#L4778-L4784
But actually it returns a 3-`tuple` in all cases (L4819, L4831, L4838):
https://github.com/pytorch/pytorch/blob/918fe519a0f9bdb2dd213eaaffbcd420b7391280/torch/nn/functional.py#L4812-L4838

---

<details>
<summary>Related error logs</summary>

```log
torch/nn/functional.py:1135:51: error: Argument 2 to "_list_with_default" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:1179:51: error: Argument 2 to "_list_with_default" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:1231:52: error: Argument 2 to "_list_with_default" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:1248:52: error: Argument 2 to "_list_with_default" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:2481:28: error: Argument 1 to "_verify_batch_size" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:2527:30: error: Argument 1 to "_verify_spatial_size" has incompatible type "Size"; expected "List[int]"  [arg-type]
torch/nn/functional.py:4819:20: error: Incompatible return value type (got "Tuple[Any, Any, Any]", expected "List[Tensor]")  [return-value]
torch/nn/functional.py:4831:20: error: Incompatible return value type (got "Tuple[Tensor, Any, Any]", expected "List[Tensor]")  [return-value]
torch/nn/functional.py:4838:16: error: Incompatible return value type (got "Tuple[Tensor, Tensor, Tensor]", expected "List[Tensor]")  [return-value]
```
</details>

### Versions

master

### PR DESCRIPTION
**Pull request by clee2000**
Query for list of renabled issues in the filter test config step: switch filter test config to query for all the PR info instead of just the labels so token usage should stay the same, move code and tests related to parsing reenabled issues to filter test config step, remove old code to get PR body and commit message.  `REENABLED_ISSUES` should be a comma separated list of issue numbers to be reenabled.  

For testing: Fixes #103789
Check that 103789 shows up in list of ignored disabled issues
Sanity check that test-config labels still work

More testing via `python3 ".github/scripts/filter_test_configs.py"     --workflow "pull"     --job-name "linux-bionic-cuda12.1-py3.10-gcc9 / test (default, 4, 5, linux.4xlarge.nvidia.gpu)"     --test-matrix "{ include: [
    { config: "default", shard: 1, num_shards: 1 },
  ]}
  "     --pr-number ""     --tag ""     --event-name "push"     --schedule ""     --branch ""`
 and
 `python3 ".github/scripts/filter_test_configs.py"     --workflow "pull"     --job-name "linux-bionic-cuda12.1-py3.10-gcc9 / test (default, 4, 5, linux.4xlarge.nvidia.gpu)"     --test-matrix "{"include": [{"config": "default", "shard": 1, "num_shards": 5, "runner": "linux.g5.4xlarge.nvidia.gpu"}, {"config": "default", "shard": 2, "num_shards": 5, "runner": "linux.g5.4xlarge.nvidia.gpu"}, {"config": "default", "shard": 3, "num_shards": 5, "runner": "linux.g5.4xlarge.nvidia.gpu"}, {"config": "default", "shard": 4, "num_shards": 5, "runner": "linux.g5.4xlarge.nvidia.gpu"}, {"config": "default", "shard": 5, "num_shards": 5, "runner": "linux.g5.4xlarge.nvidia.gpu"}]}"     --pr-number "103790"     --tag ""     --event-name "pull_request"     --schedule ""     --branch ""`

### CODE REVIEW

File: .github/workflows/_linux-test.yml [hunk 1]
```diff
@@ -212,8 +198,7 @@ jobs:
             -e SHARD_NUMBER \
             -e TEST_CONFIG \
             -e NUM_TEST_SHARDS \
-            -e PR_BODY \
```
Comment (huydhn) [inline .github/workflows/_linux-test.yml:215]: Nice!  Does this just fix this issue https://github.com/pytorch/pytorch/issues/103860 now that passing the PR_BODY is not needed anymore? :)

File: .github/workflows/_rocm-test.yml [hunk 2]
```diff
@@ -130,7 +130,7 @@ jobs:
           TEST_CONFIG: ${{ matrix.config }}
           SHARD_NUMBER: ${{ matrix.shard }}
           NUM_TEST_SHARDS: ${{ matrix.num_shards }}
-          PR_BODY: ${{ github.event.pull_request.body }}
+          REENABLED_ISSUES: ${{ github.event.pull_request.body }}
```
Comment (huydhn) [inline .github/workflows/_rocm-test.yml:133]: This looks wrong as other workflow has `REENABLED_ISSUES: ${{ steps.keep-going.outputs.reenabled-issues }}`

**Review by huydhn — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
